include/fs_utils.h

```diff
@@ -89,6 +89,34 @@ constexpr uint32_t OK_IF_EXISTS = 0x1;
 
 int create_dir(const std_fs::path& path, uint32_t mode, uint32_t flags = 0x0) noexcept;
 
+// Behaves like fseek, but logs an error stating the module, byte offset,
+// filename, and strerror on failure. Returns true on success and false on
+// failure. On failure, it close the file as it's no longer in a good state.
+//
+bool check_fseek(const char* module_name, const char* filename, FILE* stream,
+                 const long long offset, const int whence);
+
+// Returns a 'check_fseek' function object that behaves like the above, with
+// the same signature as fseek. This can be used in areas where lots of
+// sequential seeks are needed.
+//
+inline auto create_check_fseek(std::string module_name, std::string filename)
```

Comment:
Suggest renaming to `make_check_fseek_func`. The word "create" gave me a pause as it has a common meaning in the context of file operations (_"Does this create a file then does some seeking? That's one weird function..._ 🤔 _"_  😄 )

Reply:
Yup, that's a nicer name 👍